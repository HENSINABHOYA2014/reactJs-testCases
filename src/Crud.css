body{
    
    background-size: cover;
    background-repeat: no-repeat;
}

a {
  text-decoration: none;
  background: rgba(0,0,0,0.6);
  color: #ccc;
  padding: .5em 1em;
  border-radius: 4px;
  cursor: pointer;
  transition: 350ms;
}

a:hover {
  background: rgba(0,0,0,0.8);
}

.modal-container {
  position: fixed;
  z-index: 5;
  width: 100vw;
  height: 100vh;
  background: rgba(0,0,0,0.5);
  display: none;
}

.modal-container.active {
  display: flex;
  justify-content: center;
  align-items: center;
}

.modal {
  background: #333;
  width: 350px;
  min-height: 150px;
  border-radius: 12px;
  padding: 25px;
  display: flex;
  gap: 25px;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  border-left: 3px solid #2ecc71;
}

.modal-content {
  display: flex;
  flex-direction: column;
  gap: 10px;
}

.modal-content p {
  margin: 0;
  text-align: center;
}

.modal-close {
  border: none;
  outline: none;
  background: #121212;
  font-size: inherit;
  padding: .5em 1.5em;
  border-radius: 6px;
  color: #fff;
  cursor: pointer;
  transition: 350ms;
}

.modal-close:hover {
  background: #232323;
}

.toast {
  position: fixed;
  top: 16px;
  right: 25px;
  width: 375px;
  background: #1d1e22;
  padding: 25px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  /*   gap: 20px; */
  border-radius: 12px;
  border-left: 3px solid #e74c3c;
  box-shadow: rgba(0, 0, 0, 0.1) 0px 20px 25px -5px,
    rgba(0, 0, 0, 0.04) 0px 10px 10px -5px;
  overflow: hidden;
  transform: translateX(calc(100% + 25px));
  transition: all 0.5s cubic-bezier(0.68, -0.55, 0.265, 1.35);
}

.toast.active {
  transform: translateX(0);
}

.toast i:first-child {
  color: #e74c3c;
  font-size: 20px;
}

.toast-text {
  margin: 0;
  font-size: 0.8125rem;
  text-transform: uppercase;
}

.toast i:last-child {
  color: #666;
  cursor: pointer;
  transition: 350ms;
}

.toast i:last-child:hover {
  color: #ccc;
}



.header {
  background-color: #1d1e22;
  border-bottom: 1px solid #444;
  padding: 20px 40px;
}

.header h2 {
  margin: 0;
}

.form {
  padding: 30px 40px;
}

.form-control {
  margin-bottom: 10px;
  padding-bottom: 20px;
  position: relative;
}

.form-control label {
  display: inline-block;
  margin-bottom: 5px;
}

.form-control input:focus {
  outline:none;
}

.form-control input {
  background-color: #acd0cb;
  border: 1px solid #888;
  border-radius: 4px;
  display: block;
  font-size: 14px;
  font-family: inherit;
  color: black;
  width: 100%;
  padding: 10px;
}

.form-control.success input {
  border-color: #2ecc71;
}

.form-control.error input {
  border-color: #e74c3c;
}

.form-control i {
  position: absolute;
  top: 40px;
  right: 10px;
  visibility: hidden;
}

.form-control.success i.fa-check-circle {
  color: #2ecc71;
  visibility: visible;
}

.form-control.error i.fa-exclamation-circle,
.form-control.error small {
  color: #e74c3c;
  visibility: visible;
}

.form-control small {
  visibility: hidden;
  position: absolute;
  bottom: 0;
  left: 0;
}
div.form-control{
  width: 450px;
}
.form button {
  background-color: #333;
  border: 2px solid #333;
  border-radius: 4px;
  color: #fff;
  display: block;
  font-family: inherit;
  font-size: 16px;
  padding: 10px;
  width: 100%;
  cursor: pointer;
  transition: 350ms;
}

.form button:hover {
  background-color: #222;
}
/*  */
.table-wrapper {
  background: #fff;
  padding: 20px 25px;
  margin: 30px 0;
border-radius: 3px;
  box-shadow: 0 1px 1px rgba(0,0,0,.05);
}
.table-title {        
padding-bottom: 15px;
background: #435d7d;
color: #fff;
padding: 16px 30px;
margin: -20px -25px 10px;
border-radius: 3px 3px 0 0;
}
.table-title h2 {
margin: 5px 0 0;
font-size: 24px;
}
.table-title .btn-group {
float: right;
}
.table-title .btn {
color: #fff;
float: right;
font-size: 13px;
border: none;
min-width: 50px;
border-radius: 2px;
border: none;
outline: none !important;
margin-left: 10px;
}
.table-title .btn i {
float: left;
font-size: 21px;
margin-right: 5px;
}
.table-title .btn span {
float: left;
margin-top: 2px;
}
table.table tr th, table.table tr td {
  border-color: #e9e9e9;
padding: 12px 15px;
vertical-align: middle;
}
table.table tr th:first-child {
width: 60px;
}
table.table tr th:last-child {
width: 100px;
}
table.table-striped tbody tr:nth-of-type(odd) {
background-color: #fcfcfc;
}
table.table-striped.table-hover tbody tr:hover {
background: #f5f5f5;
}
table.table th i {
  font-size: 13px;
  margin: 0 5px;
  cursor: pointer;
}	
table.table td:last-child i {
opacity: 0.9;
font-size: 22px;
  margin: 0 5px;
}
table.table td a {
font-weight: bold;
color: #566787;
display: inline-block;
text-decoration: none;
outline: none !important;
}
table.table td a:hover {
color: #2196F3;
}
table.table td a.edit {
  color: #FFC107;
}
table.table td a.delete {
  color: #F44336;
}
table.table td i {
  font-size: 19px;
}
table.table .avatar {
border-radius: 50%;
vertical-align: middle;
margin-right: 10px;
}
.pagination {
  float: right;
  margin: 0 0 5px;
}
.pagination li a {
  border: none;
  font-size: 13px;
  min-width: 30px;
  min-height: 30px;
  color: #999;
  margin: 0 2px;
  line-height: 30px;
  border-radius: 2px !important;
  text-align: center;
  padding: 0 6px;
}
.pagination li a:hover {
  color: #666;
}	
.pagination li.active a, .pagination li.active a.page-link {
  background: #03A9F4;
}
.pagination li.active a:hover {        
  background: #0397d6;
}
.pagination li.disabled i {
  color: #ccc;
}
.pagination li i {
  font-size: 16px;
  padding-top: 6px
}
.hint-text {
  float: left;
  margin-top: 10px;
  font-size: 13px;
}    
/* Custom checkbox */
.custom-checkbox {
position: relative;
}
.custom-checkbox input[type="checkbox"] {    
opacity: 0;
position: absolute;
margin: 5px 0 0 3px;
z-index: 9;
}
.custom-checkbox label:before{
width: 18px;
height: 18px;
}
.custom-checkbox label:before {
content: '';
margin-right: 10px;
display: inline-block;
vertical-align: text-top;
background: white;
border: 1px solid #bbb;
border-radius: 2px;
box-sizing: border-box;
z-index: 2;
}
.custom-checkbox input[type="checkbox"]:checked + label:after {
content: '';
position: absolute;
left: 6px;
top: 3px;
width: 6px;
height: 11px;
border: solid #000;
border-width: 0 3px 3px 0;
transform: inherit;
z-index: 3;
transform: rotateZ(45deg);
}
.custom-checkbox input[type="checkbox"]:checked + label:before {
border-color: #03A9F4;
background: #03A9F4;
}
.custom-checkbox input[type="checkbox"]:checked + label:after {
border-color: #fff;
}
.custom-checkbox input[type="checkbox"]:disabled + label:before {
color: #b8b8b8;
cursor: auto;
box-shadow: none;
background: #ddd;
}
/* Modal styles */
.modal .modal-dialog {
max-width: 400px;
}
.modal .modal-header, .modal .modal-body, .modal .modal-footer {
padding: 20px 30px;
}
.modal .modal-content {
border-radius: 3px;
}
.modal .modal-footer {
background: #ecf0f1;
border-radius: 0 0 3px 3px;
}
.modal .modal-title {
  display: inline-block;
}
.modal .form-control {
border-radius: 2px;
box-shadow: none;
border-color: #dddddd;
}
.modal textarea.form-control {
resize: vertical;
}
.modal .btn {
border-radius: 2px;
min-width: 100px;
}	
.modal form label {
font-weight: normal;
}	
/* button */


.btn {
  font-family: 'Noto Sans', sans-serif;
  cursor: pointer;
  box-shadow: -1px 3px 3px 0 rgba(80, 80, 80, .2);
  padding: 17px;
  border: none;
  border-radius: 30px;
  margin: 0 10px 40px;
  color: #fff;
  font-weight: bold;
  flex: 0 0 auto;
}
button.btn.btn-1{
  background-color: #488cac;
  font-size: 19px;
  color: white;
  width: 195px;
  height: 50px;
}

.btn-1 {
  background: #ffbbcc;
  transition: all 0.25s ease-out;
}

.btn-1:hover {
  position: relative;
  top: -3px;
  box-shadow: -6px 14px 12px 2px rgba(90, 90, 90, .12);
}
