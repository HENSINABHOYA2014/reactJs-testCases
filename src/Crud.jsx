// import { useEffect, useState } from "react";
// import handleOtherData from "./helper";

// const Crud = () => {
//     const [data, setData] = useState("");
//     const [input, setInput] = useState({
//         name: '',
//         email: '',
//         password: '',
//         city: '',
//         salery: ''
//     })

//     const [record, setRecord] = useState([]);
//     const [editId, setEditId] = useState("");


//     const handleChange = (e) => {
//         const { name, value } = e.target;
//         setInput({
//             ...input, [name]: value
//         })
//     }

//     const handleSubmit = (e) => {


//         if (editId) {
//             let ans = record.map((item) => {
//                 if (item.id == editId) {
//                     return {
//                         ...input,
//                         name: input.name,
//                         email: input.email,
//                         password: input.password,
//                         city: input.city,
//                         salery: input.salery
//                     }
//                 }
//                 return item;
//             })
//             setRecord(ans);
//             localStorage.setItem('crud', JSON.stringify(ans)); // Save updated records to localStorage
//             setEditId("");
//         } else {

//             let obj = {
//                 id: Math.floor(Math.random() * 10000),
//                 name: input.name,
//                 email: input.email,
//                 password: input.password,
//                 city: input.city,
//                 salery: input.salery

//             }
//             let data = [...record, obj];
//             setRecord(data);
//             localStorage.setItem('crud', JSON.stringify(data)); // Save updated records to localStorage
//             // localStorage.setItem('crud', JSON.stringify(data));
//             // alert("Record Succesfully Insert");

//             setInput({
//                 name: '',
//                 email: '',
//                 password: '',
//                 city: '',
//                 salery: ''
//             })
//         }

//     }
//     const deleteData = (id) => {
//         let ans = record.filter((item) => {
//             return item.id !== id;
//         })
//         setRecord(ans);
//         localStorage.setItem('crud', JSON.stringify(ans));

//     }
//     const editData = (id) => {
//         let ans = record.filter((item) => {
//             return item.id === id;
//         })
//         setInput(ans[0]);
//         setEditId(id)

//     }
//     useEffect(() => {
//         let allrecord = JSON.parse(localStorage.getItem('crud'));
//         if (allrecord === null) {
//             setRecord([]);
//         } else {
//             setRecord(allrecord);
//         }
//     }, []);

//     const handleData = () => {
//         setData("Hello")
//     }

 
//     return (
//         <>
//             <center>
//                 <input type="text" placeholder="Enter text here" name="username" id="userId" value={data} onChange={(e) => setData(e.target.value)} />

//                 {/* <button  onClick={()=>setData("updated data")}>Change Profile</button>
//             <h1>{data}</h1>  */}
//                 <button onClick={handleOtherData}>Print</button>
//                 <br />

//                 <button data-testid="btn1" onClick={handleData}>Edit Function</button>
//                 <h1>{data}</h1>

//                 <img title="AI generated image" src="https://miro.medium.com/v2/resize:fit:720/format:webp/1*c1VTPfj1eGMsC9ml3QOAlw.jpeg" alt="" />
//                 <div className="modal-container">
//                     <div className="modal" id="modal">
//                         <div className="modal-content">
//                             <p>Logged in as :</p>
//                             <p className="logged-user" id="logged-user">example@gmail.com</p>
//                         </div>
//                         <button type="button" id="modal-close" className="modal-close">OK</button>
//                     </div>
//                 </div>

//                 <div className="toast" id="toast">
//                     <i className="fas fa-exclamation-circle"></i>
//                     <p className="toast-text">Oops, Wrong email or password!</p>
//                     <i className="fas fa-close" id="close"></i>
//                 </div>

//                 <div className="container">

//                     {/* <form className="form" id="form">
//                         <div className="form-control">
//                             <label htmlFor="email">Name</label>
//                             <input type="text" placeholder="Name" name="name" onChange={handleChange} value={input.name} />
//                             <i className="fas fa-check-circle"></i>
//                             <i className="fas fa-exclamation-circle"></i>

//                         </div>

//                         <div className="form-control">
//                             <label htmlFor="email">Email</label>
//                             <input type="text" placeholder="email@email.com" name="email" onChange={handleChange} value={input.email} />
//                             <i className="fas fa-check-circle"></i>
//                             <i className="fas fa-exclamation-circle"></i>

//                         </div>

//                         <div className="form-control">
//                             <label htmlFor="password">Password</label>
//                             <input type="password" placeholder="Password" name="password" onChange={handleChange} value={input.phone} />
//                             <i className="fas fa-check-circle"></i>
//                             <i className="fas fa-exclamation-circle"></i>

//                         </div>

//                         <div className="form-control">
//                             <label htmlFor="email">City</label>
//                             <input type="text" placeholder="City" name="city" onChange={handleChange} value={input.city} />
//                             <i className="fas fa-check-circle"></i>
//                             <i className="fas fa-exclamation-circle"></i>

//                         </div>

//                         <div className="form-control">
//                             <label htmlFor="email">Salery</label>
//                             <input type="text" placeholder="Salery" name="salery" onChange={handleChange} value={input.salery} />
//                             <i className="fas fa-check-circle"></i>
//                             <i className="fas fa-exclamation-circle"></i>

//                         </div>
//                         <tr>
//                             <td></td>
//                             <td>
//                                 {
//                                     (editId) ? (<input type="button" value="Edit" onClick={() => handleSubmit()} style={{ backgroundColor: 'DimGray', color: 'white', padding: '10px 25px', fontSize: '18px', border: 'none' }} />) : <input type="button" value="Submit" onClick={() => handleSubmit()}
//                                         style={{ backgroundColor: 'SlateGray', color: 'white', padding: '10px 25px', fontSize: '18px', border: 'none' }} />
//                                 }
//                             </td>
//                         </tr>
//                     </form> */}
//                 </div>
//                 {/* <table border={2} cellPadding={10}>
//                     <tr>
//                         <td>Name</td>
//                         <td><input type="text" name="name" onChange={handleChange} value={input.name} /></td>
//                     </tr>
//                     <tr>
//                         <td>phone</td>
//                         <td><input type="text" name="phone" onChange={handleChange} value={input.phone} /></td>
//                     </tr>
//                     <tr>
//                         <td></td>
//                         <td>
//                             {
//                                 (editId) ? (<input type="button" value="Edit" onClick={() => handleSubmit()} />) : <input type="button" value="Submit" onClick={() => handleSubmit()} />
//                             }
//                         </td>
//                     </tr>
//                 </table> */}
//                 <br />
//                 <br />
//                 <table border={1} >
//                     <div className="container">
//                         <div className="table-wrapper">
//                             <div className="table-title">
//                                 <div className="row">
//                                     <div className="col-sm-6">
//                                         <h2>Manage <b>Employees</b></h2>
//                                     </div>
//                                 </div>
//                             </div>
//                             <table className="table table-striped table-hover">
//                                 <thead>
//                                     <tr>
//                                         <th>Id</th>
//                                         <th>Name</th>
//                                         <th>Email</th>
//                                         <th>Password</th>
//                                         <th>City</th>
//                                         <th>Salery</th>
//                                         <th>Action</th>
//                                     </tr>
//                                 </thead>
//                                 <tbody>
//                                     {
//                                         record.map((item) => {
//                                             const { id, name, email, password, city, salery } = item
//                                             return (
//                                                 <tr key={id}>
//                                                     <td>{id}</td>
//                                                     <td>{name}</td>
//                                                     <td>{email}</td>
//                                                     <td>{password}</td>
//                                                     <td>{city}</td>
//                                                     <td>{salery}</td>

//                                                     <td>
//                                                         <button onClick={() => deleteData(id)} style={{ backgroundColor: 'lime', border: 'none', padding: '5px', margin: '5px', color: 'white' }}>
//                                                             <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-trash3-fill" viewBox="0 0 16 16">
//                                                                 <path d="M11 1.5v1h3.5a.5.5 0 0 1 0 1h-.538l-.853 10.66A2 2 0 0 1 11.115 16h-6.23a2 2 0 0 1-1.994-1.84L2.038 3.5H1.5a.5.5 0 0 1 0-1H5v-1A1.5 1.5 0 0 1 6.5 0h3A1.5 1.5 0 0 1 11 1.5Zm-5 0v1h4v-1a.5.5 0 0 0-.5-.5h-3a.5.5 0 0 0-.5.5ZM4.5 5.029l.5 8.5a.5.5 0 1 0 .998-.06l-.5-8.5a.5.5 0 1 0-.998.06Zm6.53-.528a.5.5 0 0 0-.528.47l-.5 8.5a.5.5 0 0 0 .998.058l.5-8.5a.5.5 0 0 0-.47-.528ZM8 4.5a.5.5 0 0 0-.5.5v8.5a.5.5 0 0 0 1 0V5a.5.5 0 0 0-.5-.5Z" />
//                                                             </svg>
//                                                         </button>
//                                                         <button onClick={() => editData(id)} style={{ backgroundColor: 'green', color: 'white', border: 'none', padding: '5px' }}>
//                                                             <svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" fill="currentColor" class="bi bi-pencil-fill" viewBox="0 0 16 16">
//                                                                 <path d="M12.854.146a.5.5 0 0 0-.707 0L10.5 1.793 14.207 5.5l1.647-1.646a.5.5 0 0 0 0-.708l-3-3zm.646 6.061L9.793 2.5 3.293 9H3.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.5h.5a.5.5 0 0 1 .5.5v.207l6.5-6.5zm-7.468 7.468A.5.5 0 0 1 6 13.5V13h-.5a.5.5 0 0 1-.5-.5V12h-.5a.5.5 0 0 1-.5-.5V11h-.5a.5.5 0 0 1-.5-.5V10h-.5a.499.499 0 0 1-.175-.032l-.179.178a.5.5 0 0 0-.11.168l-2 5a.5.5 0 0 0 .65.65l5-2a.5.5 0 0 0 .168-.11l.178-.178z" />
//                                                             </svg>
//                                                         </button>
//                                                     </td>
//                                                 </tr>
//                                             )
//                                         })
//                                     }
//                                 </tbody>
//                             </table>

//                         </div>
//                     </div>




//                 </table>
//             </center>
//         </>
//     )
// }
// export default Crud;