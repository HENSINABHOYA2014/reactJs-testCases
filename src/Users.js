import React from "react";
import { render } from "@testing-library/react";

export default class Users extends React.Component {

    getUserList(a,b) {
        return a+b;
    }
    render() {
        return (<>
            <h2>Users Class Component</h2>
        </>)
    }
}