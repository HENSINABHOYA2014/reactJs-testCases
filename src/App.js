import logo from './logo.svg';
import './App.css';
import './Crud.css';
import Crud from './Crud';
import 'bootstrap/dist/css/bootstrap.min.css';
import { useState } from 'react';
import 'bootstrap-icons/font/bootstrap-icons.css';
import Users from './Users';
import Query from './Query';



function App() {

  return (
    <>
      <center>
        <br />
        <br />
       
        {/* <Crud />
        <h1>Class Component Method testing</h1>
        <Users />
        <h1>Function Component Method testing</h1>
        <button>Update</button>
        <h2>getByRole</h2>
        <input type='text' value="HelloWorld" /> */}
        <h1>Multiple elements and Custom Role</h1>
        <h1>Custom Role</h1>
        <button>Click 1</button>
        <button>Click 2</button>
        <label htmlFor='input1' >User Name</label>
        <input type='text' id='input1'/>
        <label htmlFor='input1'>User Age</label>
        <input type='text' id='input2'/>
        <div role='dummy'>dummy text</div>
        <br/>
        <Query/>
      </center>
    </>
  );
}

export default App;
