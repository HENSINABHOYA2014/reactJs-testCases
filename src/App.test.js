import { fireEvent, render, screen } from '@testing-library/react';
import Crud from './Crud';
import App from './App';
import renderer from 'react-test-renderer'
import Users from './Users';
import handleOtherData from './helper';

// test('renders learn react link', () => {
//   render(<Crud />);
//   const linkElement = screen.getByText(/Logged in as /i);
//   const title = screen.getAllByTitle("AI generated image")
//   expect(linkElement).toBeInTheDocument();
//   expect(title.length).toBeGreaterThan(0);
// });

// test('renders learn react link 1', () => {
//   render(<Crud />);
//   const checkInput = screen.getByText(/example@gmail.com/i);
//   expect(checkInput).toBeInTheDocument();
// });


// test('testing input feild', () => {
//   render(<Crud />);
//   const checkInput = screen.getByRole("textbox");
//   const checkInputPlaceholder = screen.getByPlaceholderText("Enter text here")
//   expect(checkInput).toBeInTheDocument();
//   expect(checkInputPlaceholder).toBeInTheDocument();
//   expect(checkInput).toHaveAttribute("name", "username");
//   expect(checkInput).toHaveAttribute("id", "userId")
//   expect(checkInput).toHaveAttribute("type", "text")
//   expect(checkInput).toHaveAttribute("value", "")


// });

// describe("Ui testing", () => {
//   test('testing input feild', () => {
//     render(<Crud />);
//     const checkInput = screen.getByRole("textbox");
//     expect(checkInput).toHaveAttribute("name", "username");
//   })
//   describe("nested Ui testing", () => {
//     test('nested testing input feild', () => {
//       render(<Crud />);
//       const checkInput = screen.getByRole("textbox");
//       expect(checkInput).toHaveAttribute("name", "username");
//     })

//   })
// })

// describe.skip("App testing", () => {
//   test('testing input feild', () => {
//     render(<Crud />);
//     const checkInput = screen.getByRole("textbox");
//     expect(checkInput).toHaveAttribute("name", "username");
//   })
// })

// test('testing onfchange event', () => {
//   render(<Crud />);
//   const input = screen.getByRole("textbox");
//   fireEvent.change(input, { target: { value: 'a' } })
//   expect(input.value).toBe("a");
// });

// test("Class Component method testing", () => {
//   render(<App />)
//   const componentData = renderer.create(<Users />).getInstance()
//   let a = "text"
//   let b = "add"
//   expect(componentData.getUserList(a + b)).toMatch(a + b)
// })

// test("testing with onclick event",()=>{
//   render(<Crud/>)
//   const btn = screen.getByRole("button", { name: /updated data/i });
//   fireEvent.click(btn);
//   expect(screen.getByText("updated data")).toBeInTheDocument();
// })

// test("Multiple elements and Custom Role & getbyrole query", () => {
//   render(<App />)
//   const btn1 = screen.getByRole("button", { name: "Click 1" });
//   const btn2 = screen.getByRole("button", { name: "Click 2" });
//   const input1 = screen.getByRole("textbox", { name: "User name" })
//   const input2 = screen.getByRole("textbox", { name: "User Age" })
//   const dv1=screen.getByRole("dummy");
//   expect(dv1).toBeInTheDocument();

//   expect(btn1).toBeInTheDocument();
//   expect(btn2).toBeInTheDocument();
//   expect(input1).toBeInTheDocument();
//   expect(input2).toBeInTheDocument();
// })

// test("getByallrole query", () => {
//   render(<App />)
//   const btns = screen.getAllByRole("button");
//   const options = screen.getAllByRole("option");

//   expect(btns).toBeInTheDocument();

//   for (let i = 0; i < btns.length; i++) {
//     expect(btns[i]).toBeInTheDocument()
//   }

//   for (let i = 0; i < options.length; i++) {
//     expect(options[i]).toBeInTheDocument()
//   }
// })
