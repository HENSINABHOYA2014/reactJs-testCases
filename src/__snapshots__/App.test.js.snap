// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot for Crud component 1`] = `
<center>
  <input
    id="userId"
    name="username"
    onChange={[Function]}
    placeholder="Enter text here"
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Print
  </button>
  <br />
  <button
    data-testid="btn1"
    onClick={[Function]}
  >
    Edit Function
  </button>
  <h1 />
  <img
    alt=""
    src="https://miro.medium.com/v2/resize:fit:720/format:webp/1*c1VTPfj1eGMsC9ml3QOAlw.jpeg"
    title="AI generated image"
  />
  <div
    className="modal-container"
  >
    <div
      className="modal"
      id="modal"
    >
      <div
        className="modal-content"
      >
        <p>
          Logged in as :
        </p>
        <p
          className="logged-user"
          id="logged-user"
        >
          example@gmail.com
        </p>
      </div>
      <button
        className="modal-close"
        id="modal-close"
        type="button"
      >
        OK
      </button>
    </div>
  </div>
  <div
    className="toast"
    id="toast"
  >
    <i
      className="fas fa-exclamation-circle"
    />
    <p
      className="toast-text"
    >
      Oops, Wrong email or password!
    </p>
    <i
      className="fas fa-close"
      id="close"
    />
  </div>
  <div
    className="container"
  />
  <br />
  <br />
  <table
    border={1}
  >
    <div
      className="container"
    >
      <div
        className="table-wrapper"
      >
        <div
          className="table-title"
        >
          <div
            className="row"
          >
            <div
              className="col-sm-6"
            >
              <h2>
                Manage 
                <b>
                  Employees
                </b>
              </h2>
            </div>
          </div>
        </div>
        <table
          className="table table-striped table-hover"
        >
          <thead>
            <tr>
              <th>
                Id
              </th>
              <th>
                Name
              </th>
              <th>
                Email
              </th>
              <th>
                Password
              </th>
              <th>
                City
              </th>
              <th>
                Salery
              </th>
              <th>
                Action
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </table>
</center>
`;
